test/test-rewrite-url.js

```diff
@@ -0,0 +1,177 @@
+                  slug: '/tab-one/child-one'
+                },
+                {
+                  title: 'Tab One Child Two',
+                  type: 'default',
+                  slug: '/tab-one/child-two'
+                },
+              ]
+            },
+            {
+              title: 'Tab Two',
+              type: 'default',
+              slug: '/tab-two',
+              _children: [
+                {
+                  title: 'Tab Two Child One',
+                  type: 'default',
+                  slug: '/tab-two/child-one'
+                },
+                {
+                  title: 'Tab Two Child Two',
+                  type: 'default',
+                  slug: '/tab-two/child-two'
+                },
+              ]
+            },
+            {
+              title: 'Products',
+              type: 'products-page',
+              slug: '/products'
+            }
+          ],
+          types: [
+            {
+              name: 'home',
+              label: 'Home'
+            },
+            {
+              name: 'default',
+              label: 'Default'
+            },
+            {
+              name: 'products',
+              label: 'Products'
+            }
+          ]
+        },
+        'products': {
+          extend: 'apostrophe-pieces',
+          name: 'product'
+        },
+        'products-pages': {
+          extend: 'apostrophe-pieces-pages'
+        },
+      },
+      afterInit: function(callback) {
+        assert(apos.modules['apostrophe-site-map']);
+        return callback(null);
+      },
+      afterListen: function(err) {
+        done();
+      }
+    });
+  });
+
+  it('insert a product for test purposes', function(done) {
+    var product = _.assign(apos.modules.products.newInstance(), {
+      title: 'Cheese',
+      slug: 'cheese'
+    });
+    apos.modules.products.insert(apos.tasks.getReq(), product, function(err) {
+      assert(!err);
+      done();
+    });
+  });
+
+  it('make sure everything is published and out of the trash for test purposes', function(done) {
+    return apos.docs.db.update({}, {
+      $set: {
+        trash: false,
+        published: true
+      }
+    }, {
+      multi: true
+    }, function(err, count) {
+      assert(!err);
+      done();
+    });
+  });
+
+  it('insert an unpublished product for test purposes', function(done) {
+    var product = _.assign(apos.modules.products.newInstance(), {
+      title: 'Rocks',
+      slug: 'rocks',
+      published: false
+    });
+    apos.modules.products.insert(apos.tasks.getReq(), product, function(err) {
+      assert(!err);
+      done();
+    });
+  });
+
+  it('should generate a suitable sitemap', function(done) {
+    this.timeout(10000);
+    get('http://localhost:7780/sitemap.xml', function(err, xml) {
+      if (err) {
+        console.error(err);
+      }
+      assert(!err);
+      assert(xml);
+      assert(xml.indexOf('<loc>https://public-site.com/</loc>') !== -1);
```

Comment:
Note this matches the rewriteUrl override.

Reply:
You could have used 
```javascript
assert(xml.includes('<loc>https://public-site.com/</loc>'))
```
That is a bit clearer I think.
And 
```javascript
assert(!xml.includes('<loc>https://public-site.com/products/rocks</loc>'))
```
For the last line.